torch/_inductor/ir.py

```diff
@@ -4191,7 +4191,7 @@ def reduction(self, name, dtype, src_dtype, reduction_type, index, value):
 
             def index_expr(self, index, dtype):
                 if isinstance(index, (int, sympy.Integer)):
-                    return ops.constant(int(index), dtype)
+                    return self._inner.constant(int(index), dtype)
```

Comment:
`self._inner` can return a different type from the global ops handler, so we need to be more consistent here.